includes/wc-payment-api/class-wc-payments-api-client.php

```diff
@@ -1919,6 +1920,13 @@ protected function check_response_for_errors( $response ) {
 					$response_code
 				);
 			} elseif ( isset( $response_body['error'] ) ) {
+				if ( isset( $response_body['error']['code'] ) && 'amount_too_large' === $response_body['error']['code'] && 'requires_capture' === $response_body['error']['payment_intent']['status'] ) {
```

Comment:
An additional check is a good one, but it needs hardening a bit. And I'd:
- move it into a dedicated local variable (so we'll get one more nested if)
- add a comment explaining that we are redacting the message and the reasoning behind

Reply:
Thanks @kalessil, moved error code and payment intent status to local variables and added comment in this commit  https://github.com/Automattic/woocommerce-payments/pull/7581/commits/6f9a3c446cd44af8382242bce9606a78b4a6b541